File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        Map<String, Integer> counter = new HashMap<>();
        for (String[] c : clothes) {
            counter.merge(c[1], 1, (o, v) -> o + 1);
        }
        int answer = 1;
        for (int v : counter.values()) {
            answer *= v + 1;
        }
        
        return answer -= 1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.py
from collections import defaultdict

def solution(clothes):
    d = defaultdict(list)
    
    for cloth, t in clothes:
        d[t].append(cloth)
    
    
    answer = 1
    for t in d:
        answer *= len(d[t]) + 1
        
    return answer - 1